hal_st/stm32fxxx/AnalogToDigitalPinStm.hpp

```diff
@@ -24,18 +25,49 @@ namespace hal
         AdcStm& adc;
     };
 
+    class AnalogToDigitalInternalTemperatureImplStm
+        : public AnalogToDigitalPinImplBase<uint16_t>
+    {
+    public:
+        explicit AnalogToDigitalInternalTemperatureImplStm(AdcStm& adc);
+        virtual ~AnalogToDigitalInternalTemperatureImplStm() = default;
```

Comment:
Why does this destructor need to be virtual?

Reply:
Same here